docs/guides/development/python/install-pytorch-ubuntu-22-04/index.md

```diff
@@ -0,0 +1,224 @@
+
+    **Arrow keys**: Use up/down to move between services
+    **Spacebar**: Toggles selection showing an asterisk [*] if selected and blank if not [ ]
+    **Tab**: Moves the cursor to the *<Ok>* or *<Cancel>* button
+    **Enter**: Confirms your selection (activates the highlighted button)
+
+If you selected **Cancel** (or skipped the restart) the upgraded packages are still installed, but the changes won't take effect until the affected services are manually restarted or the system is rebooted.
+
+{{< note >}}
+Rebooting may seem like a quick fix, but it can obscure errors and delay troubleshooting. Restart services directly unless reboot is explicitly required.
+{{< /note >}}
+
+If you skipped service restarts during the upgrade process, you can apply the updates manually using `systemctl`:
+
+```command
+    sudo systemctl restart <service-name>
+```
+
+    Replacing `<service-name>` with the actual service, such as:
+
+```command
+    sudo systemctl restart ssh
+    sudo systemctl restart samba
+    sudo systemctl restart apache2
+```
+This ensures upgraded packages are used without requiring a full reboot. However, if the installer triggers a service restart (e.g., `systemd`, `udev`, or networking), contributors should:
+
+    - Confirm the restart completed using:
+
+        `systemctl status <service>` to verify it's active/running.
+
+    - Validate basic functionality:
+
+        Run a quick health check (e.g., `curl localhost:<port>` or confirm CLI tool availability).
+
+**Summary**:
+Messages like "no containers need to be restarted" or "No VM guests are running..." confirm that no action is needed in those areas. These checks are part of `needrestart` that can be ignored unless you're working in a containerized or virtualized environment. Also, some services may show "restart deferred" messages. This means they were flagged for restart but not restarted automatically. Use `sudo systemctl restart <service>` if needed.
+
+If the service fails to restart or behaves unexpectedly, see Troubleshooting: [Restart Response for edge-case handling](guides\development\python\pytorch-install-2-8-0-ubuntu\index.md\#restart-ts).
+
+3. Install Python and pip (if not already installed). To check:
+
+```command
+    python3 --version
+    pip3 --version
+```
+
+If it isn't installed, fails or shows an older version (e.g., Python 2.x), then install or update with:
+
+```command
+    sudo apt install python3 python3-pip
+```
+
+Then re-run the above validate version steps to confirm.
+
+4. Create a Virtual Environment (Optional)
+
+To isolate dependencies and avoid system-wide installs, you can create a virtual environment by running:
+
+```command
+    python3 -m venv pytorch-env
+```
+
+Then do:
```

Comment:
If this is a new step then why not number it as 5. If it is a sub step then should it not be
4
    a.
    b.